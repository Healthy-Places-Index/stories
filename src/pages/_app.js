/* eslint-disable react/prop-types */
import React from 'react';
import { ApolloProvider, ApolloClient, InMemoryCache } from '@apollo/client';
import 'semantic-ui-css/semantic.min.css';
import 'mapbox-gl/dist/mapbox-gl.css';

// import App from 'next/app'

function MyApp({ Component, pageProps }) {
  const client = new ApolloClient({
    uri: 'http://localhost:3000/admin/api',
    cache: new InMemoryCache(),
  });

  return (
    <ApolloProvider client={client}>
      <Component {...pageProps} />
    </ApolloProvider>
  );
}

// Only uncomment this method if you have blocking data requirements for
// every single page in your application. This disables the ability to
// perform automatic static optimization, causing every page in your app to
// be server-side rendered.
//
// MyApp.getInitialProps = async (appContext) => {
//   // calls page's `getInitialProps` and fills `appProps.pageProps`
//   const appProps = await App.getInitialProps(appContext);
//
//   return { ...appProps }
// }

export default MyApp;
