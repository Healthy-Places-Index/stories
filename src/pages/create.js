import React from 'react';
import { Container, Header } from 'semantic-ui-react';

import CreateForm from '../components/CreateForm';

const Create = () => (
  <Container>
    <Header as="h1">Add Map</Header>
    <CreateForm />
  </Container>
);

export default Create;
