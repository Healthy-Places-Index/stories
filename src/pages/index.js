import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { useRouter } from 'next/router';
import { Container, Header as Heading, Icon, Image } from 'semantic-ui-react';
import withApollo from '../providers/withApollo';

import Header from '../components/Header';
import Head from '../components/Head';
import Gallery from '../components/Gallery';
import useLocale from '../hooks/useLocale';

export const Home = ({ user, data }) => {
  const { locale } = useRouter();
  const { gallery, signUp, manage, login } = useLocale();

  return (
    <div style={{ minHeight: '100vh' }}>
      <Header user={user} />
      <Head title="imagineRio Narratives" />
      <Container style={{ marginTop: 30, marginBottom: 30 }}>
        {!user && (
          <>
            <a
              href={locale === 'pt' ? '/en' : '/pt'}
              style={{ display: 'block', float: 'right', marginLeft: 20 }}
            >
              <span>
                <Icon name="flag" />
                {locale === 'pt' ? 'English Version' : 'Versão em Português'}
              </span>
            </a>
            <a
              href={`/${locale}/signup`}
              style={{ display: 'block', float: 'right', marginLeft: 20 }}
            >
              <span>
                <Icon name="signup" />
                {signUp}
              </span>
            </a>
          </>
        )}
        <a href={`/${locale}/projects`} style={{ display: 'block', float: 'right' }}>
          <span>
            <Icon name={user && user.verified ? 'map outline' : 'user circle'} />
            {user && user.verified ? manage : login}
          </span>
        </a>
        <Heading as="h1" style={{ margin: '50px 0' }}>
          {gallery}
        </Heading>
        <Gallery data={data} />
        <Image src="img/hrc-logo.png" style={{ margin: '50px 0' }} />
      </Container>
    </div>
  );
};

Home.propTypes = {
  user: PropTypes.string,
  data: PropTypes.shape({
    allProjects: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  }).isRequired,
};

Home.defaultProps = {
  user: null,
};

export default withApollo(Home);

export async function getServerSideProps({ req }) {
  const {
    data: { data },
  } = await axios.post(`${req.protocol}://${req.get('Host')}/admin/api`, {
    query: `query GetPublished {
        allProjects(where: { gallery: true }) {
          id
          title
          description
          category
          url
          tags {
            name
          }
          user {
            name
          }
          createdAt
        }
        categories: __type(name: "ProjectCategoryType") {
          values: enumValues {
            name
          }
        }
      }
    `,
  });

  let user = null;
  if (req.user) user = req.user;

  return {
    props: {
      data,
      user,
    },
  };
}
