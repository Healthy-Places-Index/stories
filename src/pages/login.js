import React, { useState, useEffect } from 'react';
import { useMutation, gql } from '@apollo/client';
import {
  Header as Heading,
  Container,
  Segment,
  Form,
  Button,
  Image,
  Message,
} from 'semantic-ui-react';
import withApollo from '../providers/withApollo';

import Header from '../components/Header';
import Head from '../components/Head';

const Login = () => {
  const AUTH_MUTATION = gql`
    mutation signin($identity: String, $secret: String) {
      authenticate: authenticateUserWithPassword(email: $identity, password: $secret) {
        item {
          id
        }
      }
    }
  `;

  const [identity, setIdentity] = useState('');
  const [secret, setSecret] = useState('');
  const [reloading, setReloading] = useState(false);
  const [error, setError] = useState(false);

  const [signIn, { loading, client }] = useMutation(AUTH_MUTATION, {
    variables: { identity, secret },
    onCompleted: async () => {
      setReloading(true);
      await client.clearStore();
      window.location.reload(true);
    },
    onError: () => {
      setError(true);
    },
  });

  const onSubmit = e => {
    e.preventDefault();
    if (!loading) {
      signIn();
    }
  };

  useEffect(() => {
    setError(false);
  }, [identity, secret]);

  return (
    <div style={{ backgroundColor: '#FAFAFA', minHeight: '100vh' }}>
      <Head title="Login" />
      <Header />
      <Container text>
        <Heading as="h1" style={{ margin: '20% 0 50px' }}>
          Welcome to imagineRio Narratives
        </Heading>
        <Segment loading={reloading}>
          <Heading as="h3" style={{ margin: '10px 0 30px' }}>
            Log in to your account
          </Heading>
          <Form method="POST" onSubmit={onSubmit}>
            <Form.Input
              name="email"
              label="Email"
              type="email"
              value={identity}
              error={error}
              onChange={e => setIdentity(e.target.value)}
            />
            <Form.Input
              name="password"
              label="Password"
              type="password"
              value={secret}
              error={error}
              onChange={e => setSecret(e.target.value)}
            />
            <a href="/reset" style={{ float: 'right', marginBottom: 15 }}>
              Forgot your password?
            </a>
            <Button type="submit" fluid primary loading={loading}>
              Login
            </Button>
            {error && (
              <Message negative>
                You could not be logged in. Please check your username and password
              </Message>
            )}
          </Form>
        </Segment>
        <Image src="img/hrc-logo.png" />
      </Container>
    </div>
  );
};

export default withApollo(Login);
