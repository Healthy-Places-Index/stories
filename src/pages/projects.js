import React from 'react';
import PropTypes from 'prop-types';
import { useQuery, gql } from '@apollo/client';
import { Container, Header, Segment, Button, Image, Icon } from 'semantic-ui-react';
import withApollo from '../lib/withApollo';

const GET_PROJECTS = gql`
  query GetProjects($user: ID!) {
    allProjects(where: { user: { id: $user } }) {
      id
      title
      description
      category
      url
      createdAt
      updatedAt
      published
      tags {
        name
      }
    }
  }
`;

const Projects = ({ user }) => {
  const { loading, error, data } = useQuery(GET_PROJECTS, { variables: { user } });

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :(</p>;

  return (
    <Container style={{ marginTop: 30, marginBottom: 30 }}>
      <Header as="h1">My Narratives / Minhas narrativas</Header>
      <Button
        content="Add Narrative / Nova narrativa"
        icon="plus"
        size="large"
        color="blue"
        as="a"
        href="/create"
      />
      <Segment.Group>
        {data.allProjects.map(proj => (
          <Segment key={proj.id}>
            {proj.url && <Image src={proj.url} floated="left" style={{ height: 60 }} />}
            <span style={{ fontWeight: 'bold', fontSize: '1.25em' }}>
              {`${proj.title} - ${proj.category}`}
            </span>
            <Button floated="right" content="Editor" icon="edit" as="a" href={`/edit/${proj.id}`} />
            <Button
              basic
              floated="right"
              content="Preview"
              icon="play"
              as="a"
              style={{ marginRight: 10 }}
              href={`/view/${proj.id}`}
            />
            <div style={{ clear: 'right' }}>
              {proj.updatedAt && (
                <span style={{ marginRight: 10 }}>
                  {`Modified: ${new Date(proj.updatedAt).toLocaleDateString()}`}
                </span>
              )}
              {proj.createdAt && (
                <span style={{ marginRight: 10 }}>
                  {`Created: ${new Date(proj.createdAt).toLocaleDateString()}`}
                </span>
              )}
              {proj.published && (
                <span>
                  <Icon name="check" />
                  Published
                </span>
              )}
            </div>
          </Segment>
        ))}
      </Segment.Group>
      <Image src="img/hrc-logo.png" />
    </Container>
  );
};

Projects.propTypes = {
  user: PropTypes.string.isRequired,
};

export default withApollo(Projects);

export async function getServerSideProps({ req }) {
  return {
    props: {
      user: req.user.id,
    },
  };
}
