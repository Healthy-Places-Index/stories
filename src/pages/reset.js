import React, { useState } from 'react';
import axios from 'axios';
import {
  Header as Heading,
  Container,
  Segment,
  Form,
  Button,
  Image,
  Message,
} from 'semantic-ui-react';
import withApollo from '../providers/withApollo';
import useLocale from '../hooks/useLocale';

import Header from '../components/Header';
import Head from '../components/Head';

const Reset = () => {
  const { reset, resetInstructions, email: emailText, resetPassword, resetSuccess } = useLocale();
  const [email, setEmail] = useState('');
  const [error, setError] = useState('');
  const [loading, setLoading] = useState(false);
  const [success, setSuccess] = useState(false);

  const onSubmit = e => {
    e.preventDefault();
    if (email) {
      setLoading(true);
      return axios.post('/reset', { email }).then(() => {
        setLoading(false);
        setSuccess(true);
      });
    }
    return setError('Please fill out all required fields');
  };

  return (
    <div style={{ backgroundColor: '#FAFAFA', minHeight: '100vh' }}>
      <Head title="Reset your password" />
      <Header />
      <Container text>
        <Heading as="h1" style={{ margin: '50px 0' }}>
          {reset}
        </Heading>
        <Segment loading={loading}>
          <Heading as="h3" style={{ margin: '10px 0 30px' }}>
            {resetInstructions}
          </Heading>
          <Form method="POST" onSubmit={onSubmit}>
            <Form.Input
              required
              name="email"
              label={emailText}
              type="email"
              value={email}
              error={error && !email ? 'Email is required' : null}
              onChange={e => setEmail(e.target.value)}
            />
            <Button type="submit" fluid primary loading={loading}>
              {resetPassword}
            </Button>
            {error && <Message negative>{error}</Message>}
            {success && <Message success>{resetSuccess}</Message>}
          </Form>
        </Segment>
        <Image src="img/hrc-logo.png" />
      </Container>
    </div>
  );
};

export default withApollo(Reset);
