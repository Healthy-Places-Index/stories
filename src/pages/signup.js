import React, { useState } from 'react';
import { useMutation, gql } from '@apollo/client';
import {
  Header as Heading,
  Container,
  Segment,
  Form,
  Button,
  Image,
  Message,
} from 'semantic-ui-react';
import withApollo from '../providers/withApollo';

import Header from '../components/Header';
import Head from '../components/Head';

const ADD_USER = gql`
  mutation AddUser($name: String, $email: String, $password: String, $institution: String) {
    createUser(
      data: { name: $name, email: $email, password: $password, institution: $institution }
    ) {
      id
    }
  }
`;

const Signup = () => {
  const [data, setData] = useState({});
  const [error, setError] = useState('');

  const [signUp, { loading, client }] = useMutation(ADD_USER, {
    variables: { ...data },
    onCompleted: async () => {
      await client.clearStore();
      window.location.reload(true);
    },
    onError: () => {
      setError('Could not create your account. Please try again.');
    },
  });

  const onSubmit = e => {
    e.preventDefault();
    if (data.name && data.email && data.password && data.password === data.confirm) {
      signUp();
    } else {
      setError('Please fill out all required fields');
    }
  };

  return (
    <div style={{ backgroundColor: '#FAFAFA', minHeight: '100vh' }}>
      <Head title="Signup" />
      <Header />
      <Container text>
        <Heading as="h1" style={{ margin: '50px 0' }}>
          Create your account for imagineRio Narratives
        </Heading>
        <Segment loading={loading}>
          <Heading as="h3" style={{ margin: '10px 0 30px' }}>
            Enter your details below:
          </Heading>
          <Form method="POST" onSubmit={onSubmit}>
            <Form.Input
              required
              name="name"
              label="Full Name"
              type="text"
              value={data.text}
              error={error && !data.name ? 'Name is required' : null}
              onChange={e => setData({ ...data, name: e.target.value })}
            />
            <Form.Input
              required
              name="email"
              label="Email"
              type="email"
              value={data.email}
              error={error && !data.email ? 'Email is required' : null}
              onChange={e => setData({ ...data, email: e.target.value })}
            />
            <Form.Input
              name="institution"
              label="Institution"
              type="text"
              value={data.institution}
              onChange={e => setData({ ...data, institution: e.target.value })}
            />
            <Form.Group widths="equal">
              <Form.Input
                required
                name="password"
                label="Password"
                type="password"
                value={data.password}
                error={error && !data.password ? 'Password is required' : null}
                onChange={e => setData({ ...data, password: e.target.value })}
              />
              <Form.Input
                required
                name="confirm"
                label="Confirm Password"
                type="password"
                value={data.confirm}
                error={error && data.password !== data.confirm ? 'Passwords do not match' : null}
                onChange={e => setData({ ...data, confirm: e.target.value })}
              />
            </Form.Group>
            <Button type="submit" fluid primary loading={loading}>
              Create Account
            </Button>
            {error && <Message negative>{error}</Message>}
          </Form>
        </Segment>
        <Image src="img/hrc-logo.png" />
      </Container>
    </div>
  );
};

export default withApollo(Signup);
