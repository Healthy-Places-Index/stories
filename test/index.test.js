import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import { Home, GET_PROJECTS } from '../src/pages/index';

const mocks = [
  {
    request: {
      query: GET_PROJECTS,
    },
    result: {
      data: {
        allProjects: [
          {
            id: '1',
            title: 'Test Map',
            description: 'Description',
            category: 'Literature',
            url: 'https://rio-narrative-files.s3.amazonaws.com/',
            tags: [
              {
                name: 'Test',
              },
            ],
            user: {
              name: 'David Heyman',
            },
          },
        ],
      },
    },
  },
];

jest.mock('semantic-ui-react', () => {
  const moduleMock = jest.requireActual('semantic-ui-react');
  return {
    ...moduleMock,
    Popup: () => 'Popup',
  };
});

describe('home page', () => {
  it('matches snapshot', async () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Home />
      </MockedProvider>
    );

    await act(() => new Promise(resolve => setTimeout(resolve, 0)));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
