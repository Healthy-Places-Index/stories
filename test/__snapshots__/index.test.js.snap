// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "minHeight": "100vh",
    }
  }
>
  <div
    className="header"
  >
    <div
      className="ui header title"
    >
      imagineRio Narratives
    </div>
    <a
      className="logout"
      href="/projects"
    >
      Login
    </a>
  </div>
  <div
    className="ui container"
    style={
      Object {
        "marginBottom": 30,
        "marginTop": 30,
      }
    }
  >
    <h1
      className="ui header"
      style={
        Object {
          "margin": "50px 0",
        }
      }
    >
      Map Gallery
    </h1>
    <div
      className="ui three cards"
    >
      <a
        className="ui card"
        href="/view/1"
        onClick={[Function]}
      >
        <img
          className="ui image"
          src="https://rio-narrative-files.s3.amazonaws.com/"
        />
        <div
          className="content"
        >
          <div
            className="header"
          >
            Test Map
          </div>
          <div
            className="meta"
          >
            Literature
          </div>
          <div
            className="description"
          >
            David Heyman
          </div>
        </div>
        <div
          className="extra content"
        >
          Popup
        </div>
      </a>
    </div>
    <img
      className="ui image"
      src="img/hrc-logo.png"
      style={
        Object {
          "margin": "50px 0",
        }
      }
    />
  </div>
</div>
`;
